corehq/apps/reports/util.py

```diff
@@ -556,56 +557,46 @@ def _get_hq_group_id(session):
             server=TableauServer.objects.get(domain=domain)
         )]
         web_users_names = [web_user.username for web_user in WebUser.by_domain(domain)]
         # If there's a web user that isn't in the TableauUser model, create a new Tableau user
         for web_user_name in web_users_names:
             if web_user_name not in tableau_user_names:
-                _add_tableau_user_local(domain, web_user_name)
+                _add_tableau_user_local(session, web_user_name)
         # If there's a TableauUser with no corresponding WebUser, delete the Tableau user
         for tableau_user_name in tableau_user_names:
             if tableau_user_name not in web_users_names:
-                _delete_user_local(domain, tableau_user_name)
-
-    def _sync_tableau_users_with_remote(domain):
-        session = TableauAPISession.create_session_for_domain(domain)
+                _delete_user_local(session, tableau_user_name)
+
+    def _sync_tableau_users_with_remote(session):
+        # Setup
+        def _get_HQ_group_users(session):
+            remote_HQ_group_id = _get_hq_group_id(session)
+            if remote_HQ_group_id:
+                remote_HQ_group_users = session.get_users_in_group(remote_HQ_group_id)
+            else:
+                session.create_group(HQ_TABLEAU_GROUP_NAME)
+                remote_HQ_group_users = []
+            return remote_HQ_group_users
 
-        # Setup - parse/get remote group ID and users in group
-        remote_HQ_group_id = _get_hq_group_id(session)
-        if remote_HQ_group_id:
-            remote_HQ_group_users = session.get_users_in_group(remote_HQ_group_id)
-        else:
-            remote_HQ_group_id = session.create_group(HQ_TABLEAU_GROUP_NAME, "Viewer")
-            remote_HQ_group_users = []
+        all_remote_users = session.get_users_on_site()
         local_users = TableauUser.objects.filter(server=session.tableau_connected_app.server)
+        remote_HQ_group_users = _get_HQ_group_users(session)
 
         # Add/delete/update remote users to match with local reality
-        def _add_new_user_to_HQ_group(session, local_user):
-            new_user_id = _add_tableau_user_remote(session, local_user, local_user.role)
-            session.add_user_to_group(new_user_id, remote_HQ_group_id)
         for local_user in local_users:
-            remote_user = session.get_user_on_site(tableau_username(local_user.username))
-            if not remote_user:
-                _add_new_user_to_HQ_group(session, local_user)
-            elif local_user.tableau_user_id != remote_user['id']:
-                _delete_user_remote(session, remote_user['id'])
-                _add_new_user_to_HQ_group(session, local_user)
-            elif local_user.role != remote_user['siteRole']:
+            local_tableau_username = tableau_username(local_user.username)
+            if local_tableau_username not in all_remote_users.keys():
```

Comment:
nit: you can check for dict membership directly.  I'm not sure if calling `.keys()` is computationally or memory expensive, but in this case you can safely skip it, so that seems easiest.
```suggestion
            if local_tableau_username not in all_remote_users:
```